datahub-web-react/src/app/ingest/source/builder/RecipeForm/utils.tsx

```diff
@@ -1,610 +1,96 @@
-};
-
-const dashboardAllowFieldPath = 'source.config.dashboard_pattern.allow';
-export const DASHBOARD_ALLOW: RecipeField = {
-    name: 'dashboard_pattern.allow',
-    label: 'Allow Patterns',
-    tooltip: 'Use regex here.',
-    type: FieldType.LIST,
-    buttonLabel: 'Add pattern',
-    fieldPath: dashboardAllowFieldPath,
-    rules: null,
-    section: 'Dashboards',
-    setValueOnRecipeOverride: (recipe: any, values: string[]) =>
-        setListValuesOnRecipe(recipe, values, dashboardAllowFieldPath),
-};
-
-const dashboardDenyFieldPath = 'source.config.dashboard_pattern.deny';
-export const DASHBOARD_DENY: RecipeField = {
-    name: 'dashboard_pattern.deny',
-    label: 'Deny Patterns',
-    tooltip: 'Use regex here.',
-    type: FieldType.LIST,
-    buttonLabel: 'Add pattern',
-    fieldPath: dashboardDenyFieldPath,
-    rules: null,
-    section: 'Dashboards',
-    setValueOnRecipeOverride: (recipe: any, values: string[]) =>
-        setListValuesOnRecipe(recipe, values, dashboardDenyFieldPath),
-};
+import { KAFKA } from '../../conf/kafka/kafka';
+import {
+    INCLUDE_LINEAGE,
+    PROFILING_ENABLED,
+    STATEFUL_INGESTION_ENABLED,
+    DATABASE_ALLOW,
+    DATABASE_DENY,
+    UPSTREAM_LINEAGE_IN_REPORT,
+    TABLE_LINEAGE_MODE,
+    INGEST_TAGS,
+    INGEST_OWNER,
+    DASHBOARD_ALLOW,
+    DASHBOARD_DENY,
+    CHART_ALLOW,
+    CHART_DENY,
+    GITHUB_INFO_REPO,
+    EXTRACT_USAGE_HISTORY,
+    EXTRACT_OWNERS,
+    SKIP_PERSONAL_FOLDERS,
+    TOPIC_ALLOW,
+    TOPIC_DENY,
+} from './common';
+import {
+    SNOWFLAKE_ACCOUNT_ID,
+    SNOWFLAKE_WAREHOUSE,
+    SNOWFLAKE_USERNAME,
+    SNOWFLAKE_PASSWORD,
+    SNOWFLAKE_ROLE,
+    SNOWFLAKE_SCHEMA_ALLOW,
+    SNOWFLAKE_SCHEMA_DENY,
+    SNOWFLAKE_TABLE_ALLOW,
+    SNOWFLAKE_TABLE_DENY,
+    SNOWFLAKE_VIEW_ALLOW,
+    SNOWFLAKE_VIEW_DENY,
+} from './snowflake';
+import {
+    BIGQUERY_PROJECT_ID,
+    BIGQUERY_CREDENTIAL_PROJECT_ID,
+    BIGQUERY_PRIVATE_KEY,
+    BIGQUERY_PRIVATE_KEY_ID,
+    BIGQUERY_CLIENT_EMAIL,
+    BIGQUERY_CLIENT_ID,
+    BIGQUERY_SCHEMA_ALLOW,
+    BIGQUERY_SCHEMA_DENY,
+    BIGQUERY_TABLE_ALLOW,
+    BIGQUERY_TABLE_DENY,
+    BIGQUERY_VIEW_ALLOW,
+    BIGQUERY_VIEW_DENY,
+} from './bigquery';
+import {
+    REDSHIFT_HOST_PORT,
+    REDSHIFT_DATABASE,
+    REDSHIFT_USERNAME,
+    REDSHIFT_PASSWORD,
+    REDSHIFT_SCHEMA_ALLOW,
+    REDSHIFT_SCHEMA_DENY,
+    REDSHIFT_TABLE_ALLOW,
+    REDSHIFT_TABLE_DENY,
+    REDSHIFT_VIEW_ALLOW,
+    REDSHIFT_VIEW_DENY,
+} from './redshift';
+import { TABLEAU_CONNECTION_URI, TABLEAU_PROJECT, TABLEAU_SITE, TABLEAU_USERNAME, TABLEAU_PASSWORD } from './tableau';
+import { LOOKER_BASE_URL, LOOKER_CLIENT_ID, LOOKER_CLIENT_SECRET } from './looker';
+import {
+    KAFKA_SASL_USERNAME,
+    KAFKA_SASL_PASSWORD,
+    KAFKA_BOOTSTRAP,
+    KAFKA_SCHEMA_REGISTRY_URL,
+    KAFKA_SCHEMA_REGISTRY_USER_CREDENTIAL,
+    KAFKA_SECURITY_PROTOCOL,
+} from './kafka';
 
 export const RECIPE_FIELDS = {
```

Comment:
One final ask (I didn't ask you do to this originally) but now that this constant has some optional fields and is getting pretty complex, let's make a typescript interface for it. Then you can go `export const RECIPE_FIELDS: RecipeFields = { ...` and then elsewhere we know what types we're working with when we import `RECIPE_FIELDS`